bazaar/src/lib.rs

```diff
@@ -47,39 +47,38 @@ pub trait Config: frame_system::Config + encointer_communities::Config {
 
 decl_storage! {
     trait Store for Module<T: Config> as Bazaar {
-        // Maps the shop or article owner to the respective items
-        pub ShopsOwned get(fn shops_owned): double_map hasher(blake2_128_concat) CommunityIdentifier, hasher(blake2_128_concat) T::AccountId => Vec<ShopIdentifier>;
-        pub ArticlesOwned get(fn articles_owned): double_map hasher(blake2_128_concat) CommunityIdentifier, hasher(blake2_128_concat) T::AccountId => Vec<ArticleIdentifier>;
-        // Item owner
-        pub ShopOwner get(fn shop_owner): double_map hasher(blake2_128_concat) CommunityIdentifier, hasher(blake2_128_concat) ShopIdentifier => T::AccountId;
-        pub ArticleOwner get(fn article_owner): double_map hasher(blake2_128_concat) CommunityIdentifier, hasher(blake2_128_concat) ArticleIdentifier => (T::AccountId, ShopIdentifier);
-        // The set of all shops and articles per community
-        pub ShopRegistry get(fn shop_registry): map hasher(blake2_128_concat) CommunityIdentifier => Vec<ShopIdentifier>;
-        pub ArticleRegistry get(fn article_registry): map hasher(blake2_128_concat) CommunityIdentifier => Vec<ArticleIdentifier>;
+        pub BusinessRegistry get(fn business_registry): double_map hasher(blake2_128_concat) CommunityIdentifier, hasher(blake2_128_concat) T::AccountId => BusinessData;
+        pub OfferingRegistry get(fn offering_registry): double_map hasher(blake2_128_concat) BusinessIdentifier<T::AccountId>, hasher(blake2_128_concat) OfferingIdentifier => OfferingData;
```

Comment:
This is not what we had agreed on in our design thread. The idea was to have the CommunityIdentifier in the primary prefix together with BusinessIdentifier in a tuple

Reply:
Yes I agree, you had it included before. Why was it removed?

Reply:
Maybe I'm misunderstanding the issue, but to me "BusinessAccountId is the public key of an anonymous proxy aka the business. The described tuples/triples are wrapped in value type objects." (copied from readme) indicates, that the tuple is to be implemented using a proper struct ?

Reply:
Businessidentifier is a struct consisting of AcvountId and CommunityIdentifier

Reply:
Ah yes, I forgot about that change, although I agree that the naming is not ideal. 

Structs and tuple are codec-wise identical. However, I generally vote for strongly typed interfaces. So, I favor struct over tuple, as it is as expressive as possible.

Reply:
ah, sorry, should've read type definition properly. I'm not against using a struct here. But as we're only using this as an index into a double map, I don't think it is bad practise to use a tuple.

Reply:
I don't think it really matters either.